lib/cli/collect-files.js

```diff
@@ -52,6 +53,22 @@ module.exports = ({
     }
   }, []);
 
+  // check that each file passed in to --file exists
+  fileArgs.forEach(file => {
+    const fileAbsolutePath = path.resolve(file);
+    try {
+      require.resolve(fileAbsolutePath);
+    } catch (err) {
+      // this error doesn't bubble up to the middleware like how
+      // --require handles it so we log a warning here and exit
+      const warningMessage = ansi.yellow(
+        `${symbols.warning} Warning: Cannot find test file "${file}" at the path "${fileAbsolutePath}"`
+      );
+      console.warn(`${warningMessage}\n`);
+      process.exit(1);
```

Comment:
[Refactor] I don't love having a `process.exit` deep within code. [`n/no-process-exit`](https://github.com/eslint-community/eslint-plugin-n/blob/4778ae86c398f90da13248c2d31bfc4a83cd2dea/docs/rules/no-process-exit.md) has good reasoning around it.

It's also not really a "warning" yellow if the whole thing exists. At that point it's an error (red).

I think a more future-facing, approach would be either:

* (less work, but still partially flawed) have this function throw a special error that any higher-up handler(s) know how to log nicely
* (more work, but more understandable long-term) have this function return an object explaining its results, and have higher-up handler(s) work nicely with it

I haven't looked deeply enough to know whether either or both is feasible. Before we go into a whole deep dive, what do you think about this direction?

Reply:
Good to learn about the `no-process-exit`, thanks! I think the latter where we return an object and having the higher-up handler(s) deal with it sounds like the approach here. The error wasn't handled in `run-helpers` so maybe we can add something so that bubbles up to `run.js` like the other methods?

Reply:
Yeah that approach sounds like a good one to try!